src/main/kotlin/app/revanced/patcher/fingerprint/method/impl/MethodFingerprint.kt

```diff
@@ -40,6 +58,136 @@ abstract class MethodFingerprint(
+         */
+        internal fun createMethodLookupMap(classes: Iterable<ClassDef>) {
+            fun addMethodToMapList(map: MutableMap<String, MutableList<ClassAndMethod>>,
+                                   key: String, keyValue:  ClassAndMethod) {
+                var list = map[key]
+                if (list == null) {
+                    list = LinkedList()
+                    map[key] = list
+                }
+                list += keyValue
+            }
+
+            for (classDef in classes) {
+                for (method in classDef.methods) {
+                    // Key structure is: (access)(returnType)(optional: parameter types)
+                    val accessFlagsReturnKey = method.accessFlags.toString() + method.returnType.first()
+                    val accessFlagsReturnParametersKey = buildString {
+                        append(accessFlagsReturnKey)
+                        appendSignatureKeyParameters(method.parameterTypes)
+                    }
+                    val classAndMethod = ClassAndMethod(classDef, method)
+
+                    // For signatures with no access or return type specified.
+                    allMethods += classAndMethod
+                    // Access and return type.
+                    addMethodToMapList(signatureMap, accessFlagsReturnKey, classAndMethod)
+                    // Access, return, and parameters.
+                    addMethodToMapList(signatureMap, accessFlagsReturnParametersKey, classAndMethod)
+
+                    // Look up by Strings (the fastest way to resolve).
+                    method.implementation?.instructions?.forEach { instruction ->
+                        if (instruction.opcode == Opcode.CONST_STRING || instruction.opcode == Opcode.CONST_STRING_JUMBO) {
+                            val string = ((instruction as ReferenceInstruction).reference as StringReference).string
+                            addMethodToMapList(stringMap, string, classAndMethod)
+                        }
+                    }
+
+                    // The only additional lookup that could benefit, is a map of the full class name to its methods.
+                    // This would require adding a 'class name' field to MethodFingerprint,
+                    // as currently the class name can be specified only with a custom fingerprint.
+                }
+            }
+        }
+
+        /**
+         * Resolve using map built in [createMethodLookupMap]
+         *
+         * @param logger optional logger, to record the time to resolve each fingerprint.
+         */
+        internal fun Iterable<MethodFingerprint>.resolveUsingLookupMap(context: BytecodeContext, logger : Logger = NopLogger) {
+            if (allMethods.isEmpty()) throw PatchResultError("lookup map not initialized")
+
+            for (fingerprint in this) {
+                var time = System.currentTimeMillis()
```

Comment:
This extension function measures the time, but no other does and thus introduces inconsistency across the extension functions. Either consistently measure the resolution time in every resolution function or in none.